### ISSUE TITLE
spy / stub with expectation can potentially time out because CPU locks up.

### ISSUE THREAD

**Issue by jennifer-mann** [cypress-io/cypress#18]
When you have a spy / stub and you have an expectation for the spy / stub, it can potentially time out because it locks up the CPU.

**Example**

```js
it("lists users", function() {
  return expect(cy.agents().spy(App, "execute")).to.have.been.calledWith("list:users:chosen");
});
```

**Commands UI**
![screen shot 2015-04-09 at 10 51 24 am](https://cloud.githubusercontent.com/assets/1271364/7069435/c4ddbf5a-dea6-11e4-8c59-055caec39282.png)

**Activity Monitor**
![screen shot 2015-04-09 at 10 50 45 am](https://cloud.githubusercontent.com/assets/1271364/7069439/c8d5ce5e-dea6-11e4-92ca-5e4c0e86674b.png)

**Chrome warning**
![screen shot 2015-04-09 at 10 51 06 am](https://cloud.githubusercontent.com/assets/1271364/7069444/cb754f86-dea6-11e4-9aff-72b62e3c0b54.png)

**Comment by jennifer-mann** [comment]
Another example where cypress locks up for a while when spying (and not many events were being spyed on this time)

```coffee
   it "triggers new:export:report on download", ->
      cy
        .server()
        .route(/^\/report/, "fixture:reports").as("reportsGet")
        .route(/teams/, {items: []}).as("getTeams")
        .then ->
            ## internal spec helper functions
        .wait("@reportsGet")

        .then ->
          @trigger = @agents.spy App.vent, "trigger"
        .get("[data-js='export-report']:first").click()
        .wait("@getTeams").then ->
          expect(@trigger).to.be.calledWith "new:export:report"
```

**Comment by brian-mann** [comment]
Fixed in [`0.11.0`](https://github.com/cypress-io/cypress/wiki/Changelog#0110-09132015).
